pkg/plugins/handlers.go

```diff
@@ -0,0 +1,105 @@
+	"github.com/openshift/console/pkg/serverutils"
+)
+
+type PluginsHandler struct {
+	Client             *http.Client
+	PluginsEndpointMap map[string]string
+}
+
+func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map[string]string) *PluginsHandler {
+	return &PluginsHandler{
+		Client:             client,
+		PluginsEndpointMap: pluginsEndpointMap,
+	}
+}
+
+func (p *PluginsHandler) HandlePlugins(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Unsupported method, supported methods is only GET"})
+		return
+	}
+	pluginName, pluginFilePath := parsePluginNameAndPath(r.URL.Path)
+	serviceRequestURL, err := p.getServiceRequestURL(pluginName, pluginFilePath)
+	if err != nil {
+		errMsg := err.Error()
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	serviceRequestURL.Path = path.Join(serviceRequestURL.Path, pluginFilePath)
+
+	pluginRequest, err := http.NewRequest("GET", serviceRequestURL.String(), nil)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to create GET request for %q plugin: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	resp, err := p.Client.Do(pluginRequest)
+	if err != nil {
+		errMsg := fmt.Sprintf("GET request for %q plugin failed: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		errMsg := fmt.Sprintf("GET request for %q plugin failed with %d status code", pluginName, resp.StatusCode)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, resp.StatusCode, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	// filter unwanted headers from the response
+	proxy.FilterHeaders(resp)
+	// copy headers from the plugin's server response
+	for key, value := range resp.Header {
+		for _, v := range value {
+			w.Header().Add(key, v)
+		}
+	}
+
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed sending HTTP response body from %q plugin: %v", pluginName, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+}
+
+func (p *PluginsHandler) getServiceRequestURL(pluginName, pluginFilePath string) (*url.URL, error) {
+	pluginEndpoint, ok := p.PluginsEndpointMap[pluginName]
+	if ok != true {
+		return nil, fmt.Errorf("failed to create GET request: could not get endpoint for %q plugin", pluginName)
+	}
+	serviceRequestURL, err := url.Parse(pluginEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create GET request: could not parse %q endpoint for %q plugin", pluginEndpoint, pluginName)
```

Comment:
nit: omit the "failed to create GET request", this function is not creating a GET request